debian/control

```diff
@@ -14,6 +14,6 @@ Architecture: all
 Depends: ${misc:Depends}, cockpit, cockpit-networkmanager,
  cockpit-packagekit, cockpit-sosreport, 
  cockpit-storaged, cockpit-system, cockpit-ws, python3-serial, firewalld,
- asl3, asl3-menu, asl3-update-nodelist, allmon3, vim-nox, avahi-daemon
+ asl3 (>= 3.3.0), asl3-menu, asl3-update-nodelist, allmon3, vim-nox, avahi-daemon
```

Comment:
This seems tricky to coordinate between the two packages. Just so I understand the strategy here:
1. This update gets published to asl3-pi-appliance, at which time I don't think this package will be installable, because asl3 3.3.0 hasn't been published yet
2. asl3 gets updated to 3.3.0 (although I don't see a PR for https://github.com/AllStarLink/ASL3/issues/94 yet).
3. Hopefully when doing apt install, the packages are able to install without conflict, although I am a little worried that it will attempt to install asl3 3.3.0 first as a dependency, see that it is claiming files on the filesystem that are already claimed by the currently installed asl3-pi-appliance, and then cause install conflicts. 

Could we create .deb files locally and test the whole upgrade path on a test machine first before merging this? I could perhaps help in testing this.

What other scripts/packages reference the sa818 executable? It's not in asl3-menu I hope?
Maybe there is a strategy where we update the asl3 package first, to include the same sa818 scripts just in a different location or with a slightly different name, so as not to break anything existing. Then the asl3-pi-appliance package could be updated afterwards without breaking anything, either by removing these files (or if this would create too much user confusion, maybe change them to symbolic links to the sa818 scripts that were installed by asl3)?

Just trying to offer suggestions. If you've already tested all of this, then great.

Reply:
Actually `dpkg` will solve the dependencies automatically. In this case, asl3-pi-appliance - which still needs the sa818* commands - has to depend depend on asl3 v3.3.0 or later (that's the >= 3.3.0 part obviously). If you look at this pull - https://github.com/AllStarLink/ASL3/pull/95 - the files are moved and the control file is stating that is "breaks" asl3-pi-appliance before 1.8.0 (i.e. << 1.8.0). These two things combined signal to dpkg that they have to be installed together at those versions or "better". This actually is a supported process with Debian packaging and happens all the time.

Reply:
When t his releases to the beta package channel 'apt upgrade' should "just work" as the testing.